### PR TITLE
Fix entrypoint.sh for cgroup v2 usage

### PR DESCRIPTION
**Pull request by tsde**
This PR is related to https://github.com/rancher/rke/issues/3280

The added condition could rely on the cgroup version instead of cgroup driver if this seems more natural

### CODE REVIEW

**Comment by Thiryn** [comment]
Tracked the same issue in my env to `rke-tools` after upgrading to cgroup v2. Tested this PR and it fixes the issue with kubelet restarting all the containers.

File: entrypoint.sh [hunk 1]
```diff
@@ -20,6 +20,7 @@ if [ "$1" = "kube-proxy" ]; then
 fi
 
 if [ "$1" = "kubelet" ]; then
+    CGROUPDRIVER=$(/opt/rke-tools/bin/docker info | grep -i 'cgroup driver' | awk '{print $3}')
     DOCKER_ROOT=$(DOCKER_API_VERSION=1.24 /opt/rke-tools/bin/docker info 2>&1  | grep -i 'docker root dir' | cut -f2 -d:)
```
Comment (brandond) [inline entrypoint.sh:24]:
```suggestion
    CGROUPDRIVER=$(/opt/rke-tools/bin/docker info -f '{{.Info.CgroupDriver}}')
    DOCKER_ROOT=$(DOCKER_API_VERSION=1.24 /opt/rke-tools/bin/docker info -f '{{.Info.DockerRootDir}}')
```

File: entrypoint.sh [hunk 2]
```diff
@@ -31,16 +32,20 @@ if [ "$1" = "kubelet" ]; then
     done
     mount --rbind /host/dev /dev
     mount -o rw,remount /sys/fs/cgroup 2>/dev/null || true
-    for i in /sys/fs/cgroup/*; do
+
+    # Only applicable to cgroup v1
```
Comment (brandond) [inline entrypoint.sh:36]: if this is only applicable to cgroup v1, do you also need to query `.Info.CgroupVersion` from the docker info?

**Review by brandond — COMMENTED** [review]
I'm not at all familiar with what's going on here, but I'm happy to review to the best of my ability.

File: entrypoint.sh (hunk 1 above)
Comment (tsde) [reply entrypoint.sh:29]: Thanks for the suggestion @brandond 
The `-f` option surely is more effective but unfortunately we can't use it as the docker binary shipped in the rke-tools image is quite old (v1.12.3): https://github.com/rancher/rke-tools/blob/master/package/Dockerfile#L5

Updating the docker binary could be a good thing but that would probably need a separate PR

File: entrypoint.sh (hunk 2 above)
Comment (tsde) [reply entrypoint.sh:41]: Related to my previous comment, the `-f` option is not available with the docker binary shipped in the image.

File: entrypoint.sh (hunk 1 above)
Comment (brandond) [reply entrypoint.sh:29]: Docker 1.12.3 is like 7 years old! It was released 2016-10-26 and is undoubtedly riddled with vulnerabilities. Now is probably a good time to update that - perhaps in another commit in this same PR?

cc @macedogm - if we're not scanning the rke-tools stuff for vulns, we should start doing so immediately. I see a bunch of very old versions in use in this repo.

Comment (brandond) [reply entrypoint.sh:29]: also cc @snasovich on this thread

Comment (macedogm) [reply entrypoint.sh:29]: Hi @brandond, we are scanning rke-tools and also aware of this. We've an internal issue open to track this, as well as a big PR https://github.com/rancher/rke-tools/pull/161 to bump a bunch of dependencies.

Hi @snasovich, could we get a second colleague from team/area2 to help review the PR and perhaps merge it, please?

CC @pjbgf

**Comment by superseb** [comment]
@tsde The dependency bump PR was merged, there is now a more recent Docker binary available

**Comment by snasovich** [comment]
@tsde , would you be able to update this PR to incorporate changes made possible by new Docker binary?

**Comment by tsde** [comment]
Hi @superseb @snasovich 
Just rebased and applied the suggested changes using the `-f` option. I tested it on my cgroup v2 system and this looks good.

**Review by brandond — APPROVED** [review]

File: entrypoint.sh (hunk 1 above)
Comment (pjbgf) [reply entrypoint.sh]: FYI: the Docker binary has now been bumped to `20.10.24` on [rke-tools:v0.1.91](https://github.com/rancher/rke-tools/releases/tag/v0.1.91).

**Pull request merged**